pandas/core/frame.py

```diff
@@ -5114,6 +5114,8 @@ def combine(self, other, func, fill_value=None, overwrite=True):
             # see if we need to be represented as i8 (datetimelike)
             # try to keep us at this dtype
             needs_i8_conversion_i = needs_i8_conversion(new_dtype)
+            # Is this third argument documented? The docs say func is binary
```

Comment:
I was a little confused by all this. Opened https://github.com/pandas-dev/pandas/issues/23079. I think that should be resolved, and then these workarounds can hopefully be removed.